ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/KVTest.scala

```diff
@@ -276,4 +257,152 @@ object KVTest {
       }
     } yield result
 
+  private[kvutils] val metrics = new Metrics(new MetricRegistry)
+
+  private def submit(submission: DamlSubmission): KVTest[(DamlLogEntryId, DamlLogEntry)] =
+    for {
+      testState <- get[KVTestState]
+      entryId <- freshEntryId
+      (logEntry, newState) = keyValueCommitting.processSubmission(
+        entryId = entryId,
+        recordTime = testState.recordTime,
+        defaultConfig = testState.defaultConfig,
+        submission = submission,
+        participantId = testState.participantId,
+        inputState = submission.getInputDamlStateList.asScala.map { key =>
+          key -> testState.damlState.get(key)
+        }.toMap,
+      )
+      _ <- addDamlState(newState)
+    } yield {
+      // Verify that all state touched matches with "submissionOutputs".
+      assert(
+        newState.keySet subsetOf keyValueCommitting.submissionOutputs(submission)
+      )
+      // Verify that we can always process the log entry
+      val _ = KeyValueConsumption.logEntryToUpdate(entryId, logEntry)
+
+      entryId -> logEntry
+    }
+
+  private def preExecute(
+      damlSubmission: DamlSubmission): KVTest[(DamlLogEntryId, PreExecutionResult)] =
+    for {
+      testState <- get[KVTestState]
+      entryId <- freshEntryId
+      inputKeys = damlSubmission.getInputDamlStateList.asScala
+      preExecutionResult @ PreExecutionResult(
+        readSet,
+        successfulLogEntry,
+        newState,
+        outOfTimeBoundsLogEntry,
+        _,
+        _,
+        _) = keyValueCommitting
+        .preExecuteSubmission(
+          defaultConfig = testState.defaultConfig,
+          submission = damlSubmission,
+          participantId = testState.participantId,
+          inputState = inputKeys.map { key =>
```

Comment:
To improve readability, please move this transformation logic into a helper function.

Reply:
Done.